tools/MicrosoftConfigurationBuilders.settings.targets

```diff
@@ -1,59 +1,90 @@
+    <NuGetPackageVersion>$(VersionMajor).$(VersionMinor).$(VersionRelease)</NuGetPackageVersion>
   </PropertyGroup>
 
-  <!-- Default properties -->
-
-  <PropertyGroup>
-    <RepositoryRoot Condition="'$(RepositoryRoot)' == ''">$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), MicrosoftConfigurationBuilders.sln))\</RepositoryRoot>
-    <SolutionDir>$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), MicrosoftConfigurationBuilders.sln))\</SolutionDir>
+  <PropertyGroup Condition="'$(MSBuildProjectExtension)' == '.nuproj'">
+    <NuGetPackageVersion Condition="'$(UpdateNightlyPackages)' == 'true'">$(NuGetPackageVersion)-b$(VersionBuild)</NuGetPackageVersion>
   </PropertyGroup>
 
+  <!-- Default properties -->
+
   <PropertyGroup Label="Common Configuration">
     <Configuration Condition="'$(Configuration)' == ''">Release</Configuration>
-    <OutputPath>$(RepositoryRoot)bin\$(Configuration)\</OutputPath>
-    <IntermediateOutputPath>$(RepositoryRoot)obj\$(Configuration)\$(MSBuildProjectName)\</IntermediateOutputPath>
-  </PropertyGroup>
-
-  <PropertyGroup>
+    <OutputPath>$(RepositoryRoot).binaries\bin\$(Configuration)\</OutputPath>
+    <IntermediateOutputRoot>$(RepositoryRoot).binaries\obj\$(Configuration)\</IntermediateOutputRoot>
+    <IntermediateOutputPath>$(IntermediateOutputRoot)$(MSBuildProjectName)\</IntermediateOutputPath>
+    <AssemblyPath Condition="'$(CodeSignEnabled)' == 'true'">$(CodeSignOutputPath)</AssemblyPath>
     <AssemblyPath Condition="'$(AssemblyPath)' == ''">$(OutputPath)</AssemblyPath>
-    <NuGetOutputPath>$(AssemblyPath)Packages</NuGetOutputPath>
-    <NuGetSymbolsOutputPath>$(AssemblyPath)\SymbolPackages</NuGetSymbolsOutputPath>
-    <TestOutputPath>$(OutputPath)test\</TestOutputPath>
   </PropertyGroup>
 
-  <PropertyGroup>
-    <ReferencePackagesPath>$(RepositoryRoot)packages\</ReferencePackagesPath>
+  <PropertyGroup Label="Nuget Configuration">
+    <NuGetOutputPath>$(RepositoryRoot).binaries\Packages\$(Configuration)</NuGetOutputPath>
     <NuGetPackSymbols Condition="'$(NuGetPackSymbols)' == ''">true</NuGetPackSymbols>
+    <ReferencePackagesPath>$(RepositoryRoot)packages\</ReferencePackagesPath>
     <SourceRootFullPath>$(RepositoryRoot)\src\$(MSBuildProjectName)\</SourceRootFullPath>
   </PropertyGroup>
 
   <PropertyGroup>
-    <CustomAfterMicrosoftCommonTargets>$(RepositoryRoot)tools\MicrosoftConfigurationBuilders.targets</CustomAfterMicrosoftCommonTargets>
-    <CustomAfterNuGetProjTargets>$(CustomAfterMicrosoftCommonTargets)</CustomAfterNuGetProjTargets>
+    <TestRoot>$(RepositoryRoot)\test\Microsoft.Configuration.ConfigurationBuilders.Test</TestRoot>
+    <BuildingTestProject Condition="$(MSBuildProjectDirectory.ToLower().Contains($(TestRoot.ToLower())))">true</BuildingTestProject>
   </PropertyGroup>
+
+  <Target Name="PreventReferenceCopy" AfterTargets="ResolveAssemblyReferences">
```

Comment:
This keeps the output directory clean of dependencies, so I can run signature verification on just the assemblies built by this solution. Otherwise all the extra libraries get dumped in the bin folder as well - which would make sense if this were a deployable application project. But it's not, so I added this to make the build process cleaner.